cmd/docker.go

```diff
@@ -10,49 +10,44 @@ import (
 // DockerFlags holds the flags for the docker command
 type DockerFlags struct {
 	DisableTty bool
+	EnvVariables []string
 }
 
 var dockerCmd = &cobra.Command{
 	Use:                "docker [options] [image] [command]",
 	Args:               cobra.MinimumNArgs(1),
 	Run:                runDocker,
-	DisableFlagParsing: true,
 	Short:              "Creates a new container and runs the command in it.",
 	Long: `This command acts as a helper for docker run.
 You can start with options that go before the image name
 for docker run itself, i.e --env='VAR=VALUE'. Then you must pass
 the image name and the command you want to exucute on that image.`,
 }
 
-var dockerFlags = &DockerFlags{false}
+var dockerFlags = &DockerFlags{false, []string{}}
 
 func init() {
 	rootCmd.AddCommand(dockerCmd)
 
 	dockerCmd.Flags().BoolVarP(&dockerFlags.DisableTty, "disable-tty", "T", false, "Disables TTY")
-}
+	dockerCmd.Flags().StringArrayVarP(&dockerFlags.EnvVariables, "env", "e", []string{}, "Environment variables")
 
-func runDocker(cmd *cobra.Command, originalArgs []string) {
-	var (
-		args []string
-	)
+	//After a non-flag arg, stop parsing flags
+	dockerCmd.Flags().SetInterspersed(false)
```

Comment:
Nice!

Reply:
So this removes the need for `DisableFlagParsing` which is nice. But in the other hand, we are only now able to set the --env option for `docker run`, and not any others like i.e `-p 80:80`. I assume that is OK for now, although it decreases the flexibility, I see more benefit on improving how we handle the flags at this moment.